engine/src/flutter/shell/common/shell_io_manager.cc

```diff
@@ -51,11 +51,11 @@ ShellIOManager::ShellIOManager(
           std::move(unref_queue_task_runner),
           unref_queue_drain_delay,
           resource_context_,
-          /*drain_immediate=*/!!impeller_context)),
+          /*drain_immediate=*/impeller_context.valid())),
```

Comment:
Saw some assertion errors during app shutdown related to this.  This is also causing some of the unit test failures.

The `SkiaUnrefQueue` needs to know whether Impeller or Skia is being used.  If Skia is used, then the `SkiaUnrefQueue` needs to deference objects on a thread that can invoke graphics APIs.